internal/storage/migrations/3-migrate-payments-events-from-v2.go

```diff
@@ -0,0 +1,89 @@
+package migrations
+
+import (
+	"context"
+	"math/big"
+	"time"
+
+	"github.com/formancehq/payments/internal/models"
+	"github.com/google/uuid"
+	"github.com/uptrace/bun"
+)
+
+type v2PaymentAdjustments struct {
+	bun.BaseModel `bun:"payments.adjustment"`
+
+	ID        uuid.UUID            `bun:"id,pk,nullzero"`
+	PaymentID models.PaymentID     `bun:"payment_id,pk,nullzero"`
+	CreatedAt time.Time            `bun:"created_at,nullzero"`
+	Reference string               `bun:"reference"`
+	Amount    *big.Int             `bun:"amount"`
+	Status    models.PaymentStatus `bun:"status"`
+}
+
+func MigratePaymentsFromV2(ctx context.Context, db bun.IDB) error {
+	exist, err := isTableExisting(ctx, db, "payments", "adjustment")
+	if err != nil {
+		return err
+	}
+
+	if !exist {
+		// Nothing to migrate
+		return nil
+	}
+
+	var createdAt time.Time
+	for {
+		var adjustments []v2PaymentAdjustments
+		query := db.NewSelect().
+			Model(&adjustments).
+			Order("created_at ASC").
+			Limit(100)
+
+		if !createdAt.IsZero() {
+			query.Where("created_at > ?", createdAt)
```

Comment:
So one thing that is a bit problematic is that our timestamps aren't very precise (they are truncated at the second), so if we have two records that were inserted within a second of each other, we risk skipping some records and not importing everything due to this greater than clause.

Additionally since there is no auto-increment id field in the tables, we have no reliable way to provide a paging mechanism in the case that two or more records share the same timestamp.

Let's double check the production dataset to see if there are no two records with the same timestamp in the same table and depending on the outcome one of these two strategies maybe:

a) if there are no overlaps, then we can probably go ahead of this migration and forget about it
b) before the date of the migration, alter the table to add an autoincrement ID field and then an index on this field so that we can be sure not to skip any records.

This also means that payments v3 has the same issue (no reliable way to page without potentially skipping some records in the listing endpoints) so let's make a separate ticket for that...